socket/sandboxes/heka/output/heka_tcp.lua

```diff
@@ -59,16 +59,16 @@ end
 
 local client, err = create_client()
 
-local function send_message(msg, i)
-    local len, err, i = client:send(msg, i)
-    if not len then
-        if err == "timeout" or err == "closed" then
+local function send_message(msg)
+    local i, len, err = 0
+    repeat
+        len, err, i = client:send(msg, i + 1)
+        if not len then
```

Comment:
I haven't seen a  partial send in testing or our production tests yesterday (10s of million of messages with a decent size variance) so if it even happens at all it is rare and we will just treat it as a fatal error and let HS deal with the retry.  As for the receiving end any corruption in the stream (partial message) will be logged and the data discarded.